crates/interpreter/src/syntax.rs

```diff
@@ -546,6 +548,46 @@ impl CvtOp {
     }
 }
 
+#[cfg(feature = "threads")]
+macro_rules! impl_atomic_op {
+    ($n:ident, $u:ident) => {
+        paste::paste! {
+            impl AtomicOp {
+                pub fn [< $n _ $u >](&self, mem:*mut $u, c:$u, c2:$u) -> $n {
```

Comment:
It seems to me this could be simplified by returning `$u`, reducing the combinatorics explosion. The conversion from `$u` to `$n` can be done by the caller.

Also, maybe we should handle cmpxchg separately to reduce fake arguments.